pkg/crd/spec.go

```diff
@@ -72,23 +171,26 @@ func (p *Parser) NeedCRDFor(groupKind schema.GroupKind) {
 		},
 	}
 
+	versionMap := make(map[string]apiext.CustomResourceDefinitionVersion)
 	for _, pkg := range packages {
 		typeIdent := TypeIdent{Package: pkg, Name: groupKind.Kind}
 		typeInfo := p.Types[typeIdent]
 		if typeInfo == nil {
 			continue
 		}
 		fullSchema := FlattenEmbedded(p.flattener.FlattenType(typeIdent), pkg)
-		ver := apiext.CustomResourceDefinitionVersion{
-			Name:   p.GroupVersions[pkg].Version,
+		verName := p.GroupVersions[pkg].Version
+		versionMap[verName] = apiext.CustomResourceDefinitionVersion{
+			Name:   verName,
 			Served: true,
 			Schema: &apiext.CustomResourceValidation{
 				OpenAPIV3Schema: fullSchema,
 			},
 		}
-		crd.Spec.Versions = append(crd.Spec.Versions, ver)
 	}
 
+	crd.Spec.Versions = SortCRDVersions(versionMap)
```

Comment:
you could also simplify this a bit by calling `sort.Slice` and putting parsed versions in a lookaside map, potentially.

Reply:
Using `sort.Strings` instead